File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) { 
        Map<Integer,Integer> map = new HashMap<>();
        int t;
        int[] x = new int[2];
        for(int i = 0; i < nums.length; i++) {
            t = target - nums[i];
            if (map.containsKey(t)) {
                x[0] = map.get(t);
                x[1] = i;
                return x;
            }
            map.put(nums[i],i);
        }
        return null;
    }  
}

File: 1-two-sum/1-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        dif = {}
        for i in range(len(nums)): # O(n) - iterating through list
            difference = target - nums[i]
            if nums[i] in dif: # O(1) - dictionary access
                return [i, dif[nums[i]]]
            else:
                dif[difference] = i
        